packages/bus-sqs/src/sqs-transport.ts

```diff
@@ -563,6 +596,71 @@ export class SqsTransport implements Transport<SQSMessage> {
       )
     return delay / MILLISECONDS_IN_SECONDS
   }
+
+  private async assertSnsSqsSubscriptionByArn(
+    topicArn: string,
+    sqsQueueArn: string
+  ) {
+    let nextToken = undefined
+    try {
+      do {
+        const command = new ListSubscriptionsByTopicCommand({
+          TopicArn: topicArn,
+          NextToken: nextToken
+        })
+
+        const response: ListSubscriptionsByTopicResponse = await this.sns.send(
+          command
+        )
+        const subscriptions = response.Subscriptions
+
+        if (subscriptions) {
+          for (const sub of subscriptions) {
+            return sub.Protocol === 'sqs' && sub.Endpoint === sqsQueueArn
+          }
```

Comment:
think this was meant to be:

```suggestion
          const isQueueSubscribed = subscriptions.some(sub => sub.Protocol === 'sqs' && sub.Endpoint === sqsQueueArn)
          if (isQueueSubscribed) {
            return true
          }
```